File: WEEK-6/AT-1.m
clc;
clear all;
close all;
x=[];
lenx=input("Enter length of array:");
y=0;
for i=1:lenx
  f=input("Enter x[n]:")
  x=[x,f];
end
leny=input("Enter length of y[k]:")
k=1:0.01:leny;

for i=1:lenx
  y+=x(i)*cos(2*pi*i*k);
end
plot(y);

File: WEEK-6/AT_1.py
from matplotlib import pyplot as plt
import numpy as np
x = []
y = []
lenx = int(input("Enter length of x: "))
leny = int(input("Enter length of y: "))
print("Enter elements into x:")
for i in range(lenx):
    a = int(input("Enter: "))
    x.append(a)  
print("Enter elements into y:")
for i in range(leny):
    a = int(input("Enter: "))
    y.append(a)
z = []
for k in range(leny - 1):
    s = 0
    for n in range(lenx):
        if n + k < lenx:
            s += x[n] * y[n + k]
    z.append(s)
plt.plot(z)
plt.xlabel('Lag')
plt.ylabel('Cross-correlation')
plt.title('Cross-correlation between x and y')
plt.show()
